src/client/pythonEnvironments/common/environmentIdentifier.ts

```diff
@@ -122,6 +123,103 @@ async function isWindowsStoreEnvironment(interpreterPath: string): Promise<boole
     return false;
 }
 
+/**
+ * Checks if a Pipfile associated with the provided directory exists.
+ * @param cwd the directory to look into
+ * @param lookIntoParentDirectories set to true if we should also search for Pipfile in parent directory
+ */
+function checkIfPipFileExists(cwd: string, lookIntoParentDirectories: boolean): boolean {
+    const pipFileName = process.env.PIPENV_PIPFILE;
+    let depthToSearch = 1;
+    if (lookIntoParentDirectories) {
+        // PIPENV_MAX_DEPTH tells pipenv the maximum number of directories to recursively search for
+        // a Pipfile, defaults to 3: https://pipenv.pypa.io/en/latest/advanced/#pipenv.environments.PIPENV_MAX_DEPTH
+        if (process.env.PIPENV_MAX_DEPTH) {
+            depthToSearch = +process.env.PIPENV_MAX_DEPTH;
+        } else {
+            depthToSearch = 3;
+        }
+    }
+    while (depthToSearch > 0 && cwd === path.dirname(cwd)) {
+        if (typeof pipFileName === 'string' && pathExists(path.join(cwd, pipFileName))) {
+            return true;
+        }
+        if (pathExists(path.join(cwd, 'Pipfile'))) {
+            return true;
+        }
```

Comment:
`pipFileName` will always be string if you take the above suggestion.
```suggestion
        if (await pathExists(path.join(cwd, pipFileName))) {
            return true;
        }
```

Reply:
Oops I missed to put await before pathExists call.